RFCs/FS-1135-random-functions-for-collections.md

```diff
@@ -0,0 +1,206 @@
+
+Two functions should be added to each module.
+
+```fsharp
+// Array module
+val randomSample: count:int -> array:'T[] -> 'T[]
+val randomSampleWith: random:Random -> count:int -> array:'T[] -> 'T[]
+// List module
+val randomSample: count:int -> list:'T list -> 'T list
+val randomSampleWith: random:Random -> count:int -> list:'T list -> 'T list
+// Seq module
+val randomSample: count:int -> source:'T seq -> 'T seq
+val randomSampleWith: random:Random -> count:int -> source:'T seq -> 'T seq
+```
+[ArgumentNullException](https://learn.microsoft.com/en-us/dotnet/api/system.argumentnullexception) should be raised if collection is null
+
+[ArgumentException](https://learn.microsoft.com/en-us/dotnet/api/system.argumentoutofrangeexception) should be raised if N is greater than collection length or is negative
+
+[ArgumentException](https://learn.microsoft.com/en-us/dotnet/api/system.argumentexception) should be raised if collection is empty
+
+Example:
+```fsharp
+let allPlayers = [ "Alice"; "Bob"; "Charlie"; "Dave" ]
+let round1Order = allPlayers |> List.randomSample 3 // ["Charlie", "Dave", "Alice"]
+```
+
+# Drawbacks
+
+System.Random interface has added some new methods in [.NET 8](https://learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-8#methods-for-working-with-randomness), where naming is a bit different. More new methods can eventually be added in future .NET versions.
+
+# Alternatives
+
+Use online snippets, or provide a nuget package.
+
+# Compatibility
+
+Please address all necessary compatibility questions:
+
+* Is this a breaking change? **No, unless user defined those extensions themselves**
+* What happens when previous versions of the F# compiler encounter this design addition as source code? **Should be fine**
+* What happens when previous versions of the F# compiler encounter this design addition in compiled binaries? **Should be fine**
+* If this is a change or extension to FSharp.Core, what happens when previous versions of the F# compiler encounter this construct? **Will work as usual**
+
+# Pragmatics
+
+## Diagnostics
+
+Please list the reasonable expectations for diagnostics for misuse of this feature. **I don't see a way to misuse it**
+
+## Tooling
+
+Please list the reasonable expectations for tooling for this feature, including any of these:
+
+* Debugging
+    * Breakpoints/stepping
+    * Expression evaluator
+    * Data displays for locals and hover tips
+* Auto-complete
+* Tooltips
+* Navigation and Go To Definition
+* Colorization
+* Brace/parenthesis matching
+
+**Should work just like for other collections functions**
+
+## Performance
+
+Please list any notable concerns for impact on the performance of compilation and/or generated code
+
+* For existing code **Existing code should not be affected**
+* For the new features **Performance should be respected when implementing this feature, since it can be used in performance-sensitive scenaris**
```

Comment:
nit
```suggestion
* For the new features **Performance should be respected when implementing this feature, since it can be used in performance-sensitive scenarios**
```

Reply:
Fixed